### DISCUSSION TITLE
What is the license of ripgrep?!

### DISCUSSION THREAD

**Discussion by mmahmoudian** [BurntSushi/ripgrep#2154] [Q&A] [answered]
Hi,

While trying to figure out what license I should use for my own software, i thought to explore what my favorite tools have done and ripgrep is in my top5 rust tools. Now here is the confusion: on one hand there is [a MIT license](https://github.com/BurntSushi/ripgrep/blob/0b36942f680bfa9ae88a564f2636aa8286470073/LICENSE-MIT) in the files and on the other hand there is [an unlicense](https://github.com/BurntSushi/ripgrep/blob/0b36942f680bfa9ae88a564f2636aa8286470073/UNLICENSE) file. Both files' last commit date back to Feb 27 2016 in [9d1e619](https://github.com/BurntSushi/ripgrep/commit/9d1e619ff359b6e609b02f01e36952e603104bc6) commit. At first I though this can be a dual-license software, but I've never seen such combination (inspite of [the sentence in the README](https://github.com/BurntSushi/ripgrep/blob/0b36942f680bfa9ae88a564f2636aa8286470073/README.md?plain=1#L15)).

Now here is my question: Is it even possible to have these two licenses together? The combo feels like [WTFPL](http://www.wtfpl.net/) but I'm not sure about the implications of such dual-license.

**Comment by BurntSushi** [comment, marked as answer]
The [`COPYING`](https://github.com/BurntSushi/ripgrep/blob/master/COPYING) file states that it is dual licensed under the Unlicense and the MIT.

The [FAQ](https://github.com/BurntSushi/ripgrep/blob/master/FAQ.md#license) says a bit more about the scheme.

> Is it even possible to have these two licenses together?

This is a question for a lawyer. I am not a lawyer. All I can tell you is that I've been using this licensing scheme for most of my projects for the last 7 years or so. In that time, I'm not aware of it preventing anyone from *using* my code. There is however at least one megacorp that currently forbids its employees from _contributing_ to these projects. I'd prefer not to name them openly since my goal isn't to shame them. But if you're looking to use this license combo, then it seems like material information to you.

See also:

* A frustrating exchange I had with Rick Moen: https://github.com/docopt/docopt.rs/issues/1#issuecomment-428682183
* The discussion that convinced me to move from Unlicense to Unlicense/MIT: https://github.com/BurntSushi/byteorder/issues/26

**Reply by mmahmoudian** [reply]
Wow, some of those comments are so heated that passes my tolerance level 😅 

Thanks for the explanation and the link. I came to your repos for inspiration and I think I got what I came for :)

Btw, thanks for making a tool better than silversearcher.
